src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -4293,6 +4293,24 @@ TEST_CASE("AdaptiveMeanPoolingTestCase", "[ANNLayerTest]")
   REQUIRE(arma::accu(delta) == 2.25);
 }
 
+/**
+ * Simple Separable Convolution layer test.
+ */
+TEST_CASE("SimpleSeparableConvolutionLayerTest", "[ANNLayerTest]")
+{
+  arma::mat input(224 * 224 * 3, 1), output;
+  SeparableConvolution<> module1(3, 21, 3, 3, 1, 1, 0, 0, 224, 224, 3);
+
+  // Test the forward function.
+  input.fill(2);
+  module1.Parameters().fill(1);
+  module1.Reset();
+  module1.Forward(input, output);
+
+  // Comparison value taken from PyTorch conv layer with the same config
+  REQUIRE(arma::accu(output) == 19664316);
```

Comment:
I think the plan is to compare the gradient as well?

Reply:
Should add that too.

Reply:
Um a bit confsued here but how can i get the gradients for a single layer in pytorch and repeat the same in mlpack.

Reply:
I have added all the things except this one, I am not sure how to execute it.

Reply:
You should be able to just use:

```python
layer._parameters['weight'].grad.data
layer._parameters['bias'].grad.data
```

Reply:
I believe that shows as `None` if you only have a single component, also how do you do this in mlpack.

Reply:
You can construct just a simple network, https://discuss.pytorch.org/t/how-to-print-the-computed-gradient-values-for-a-network/34179 does have an example. You can do the same thing in mlpack and use https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/ffn.hpp#L273 to get the gradient. You could implement a numerical gradient check, https://github.com/mlpack/mlpack/blob/master/src/mlpack/tests/ann_layer_test.cpp#L131-L167 has an example. In this case you just define the layer and the input.